java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -133,6 +214,20 @@ private RedisRequest.Builder prepareRedisRequest(
                                         .setArgsArray(commandArgs.build())
                                         .build());
 
+        return prepareRedisRequestRoute(builder, route);
+    }
+
+    /**
+     * Build a protobuf command's routing options.<br>
+     * Used by {@link CommandManager}.
+     *
+     * @param builder Command builder
+     * @param route Command routing parameters
+     * @return An uncompleted request. {@link CallbackDispatcher} is responsible to complete it by
+     *     adding a callback id.
+     */
+    private RedisRequest.Builder prepareRedisRequestRoute(
+            RedisRequest.Builder builder, Optional<Route> route) {
```

Comment:
Second arg could be just `Route` if you move `route.isEmpty()` check outside the function.
Up to you

Reply:
Then I would have to make another signature for ClusterTransaction calls.  This is fine.